zipline/pipeline/factors/basic.py

```diff
@@ -514,13 +514,19 @@ def _validate(self):
 
     def compute(self, today, assets, out, classifier_values):
         # Convert classifier array to group label int array
-        group_labels, _ = self.inputs[0]._to_integral(classifier_values[0])
+        group_labels, null_label = self.inputs[0]._to_integral(
+            classifier_values[0]
+        )
         _, inverse, counts = unique(  # Get counts, idx of unique groups
             group_labels,
             return_counts=True,
             return_inverse=True,
         )
-        out[:] = counts[inverse]
+        out[:] = where(
```

Comment:
I think slightly more efficient here would be:

```python
np.copyto(out, counts[inverse], group_labels != null_label)
```

```
In [9]: %timeit a[mask] = cp[mask]
The slowest run took 5.88 times longer than the fastest. This could mean that an intermediate result is being cached.
10000 loops, best of 3: 44.6 µs per loop

In [10]: %timeit np.copyto(a, cp, where=mask)
The slowest run took 5.96 times longer than the fastest. This could mean that an intermediate result is being cached.
10000 loops, best of 3: 24.2 µs per loop
```